raiden/raiden_event_handler.py

```diff
@@ -94,35 +101,39 @@ class RaidenEventHandler:
     def on_raiden_event(self, raiden: 'RaidenService', event: Event):
         # pylint: disable=too-many-branches
         if type(event) == SendLockExpired:
-            self.handle_send_lockexpired(raiden, event)
+            self.handle_send_lockexpired(raiden, cast(SendLockExpired, event))
```

Comment:
The convention that Augusto used for these kinds of checks is an `assert isinstance(X, Y), MYPY_ANNOTATION`. Please use the same approach.

Reply:
+1. We want to avoid casting. I will pick this PR up and build on it to avoid too many conflicts.

Reply:
Done here: https://github.com/raiden-network/raiden/pull/3916/commits/28ae75521fdb86caa1ee9ccba4db05c88eb33f90 and here: https://github.com/raiden-network/raiden/pull/3916/commits/0ccf4a1692969d32f20e21ab9b4d0b2faff61dc4